src/argilla/client/feedback/dataset/mixins.py

```diff
@@ -0,0 +1,296 @@
+                    f" exception: {e}"
+                ) from e
+
+        question_name2id = {}
+        questions = []
+        for question in self._questions:
+            try:
+                new_question = datasets_api_v1.add_question(
+                    client=httpx_client, id=argilla_id, question=question.dict(exclude={"id"})
+                ).parsed
+                question.id = new_question.id
+                questions.append(question)
+                question_name2id[new_question.name] = new_question.id
+            except Exception as e:
+                delete_dataset(dataset_id=argilla_id)
+                raise Exception(
+                    f"Failed while adding the question '{question.name}' to the `FeedbackDataset` in Argilla"
+                    f" with exception: {e}"
+                ) from e
+
+        try:
+            datasets_api_v1.publish_dataset(client=httpx_client, id=argilla_id)
+        except Exception as e:
+            delete_dataset(dataset_id=argilla_id)
+            raise Exception(f"Failed while publishing the `FeedbackDataset` in Argilla with exception: {e}") from e
+
+        for i in trange(
+            0, len(self.records), PUSHING_BATCH_SIZE, desc="Pushing records to Argilla...", disable=show_progress
+        ):
+            try:
+                records = []
+                for record in self.records[i : i + PUSHING_BATCH_SIZE]:
+                    if record.suggestions:
+                        for suggestion in record.suggestions:
+                            suggestion.question_id = question_name2id[suggestion.question_name]
+                    records.append(
+                        record.dict(
+                            exclude={"id": ..., "suggestions": {"__all__": {"question_name"}}}, exclude_none=True
+                        )
+                    )
+                datasets_api_v1.add_records(
+                    client=httpx_client,
+                    id=argilla_id,
+                    records=records,
+                )
+            except Exception as e:
+                delete_dataset(dataset_id=argilla_id)
+                raise Exception(
+                    f"Failed while adding the records to the `FeedbackDataset` in Argilla with exception: {e}"
+                ) from e
+
+        self.__class__ = _ArgillaFeedbackDataset
+        self.__init__(
+            client=httpx_client,
+            id=argilla_id,
+            name=name,
+            workspace=workspace,
+            fields=fields,
+            questions=questions,
+            guidelines=self.guidelines,
+        )
+        return self
+
+    @classmethod
+    def from_argilla(
+        cls: Type["FeedbackDataset"],
+        name: Optional[str] = None,
+        *,
+        workspace: Optional[str] = None,
+        id: Optional[str] = None,
+        with_records: bool = True,  # TODO(alvarobartt): deprecate `with_records`
```

Comment:
If we set to `None` as default, we can detect that a value is provided and show the warning properly.
```suggestion
        with_records: Optional[bool] = None,  # TODO(alvarobartt): deprecate `with_records`
```

Reply:
Yes, docstrings and messages have not been revisited yet, just the functionality